### PR TITLE
New module: ganon

### PR DESCRIPTION
**Pull request by jfy133**
Adds a module for [ganon](https://pirovc.github.io/ganon/) classify (note: the other sub commands produce similar-ish log files but are not informative)

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` has been updated

- [x] There is example tool output for tools in the [<https://github.com/ewels/MultiQC_TestData>](https://github.com/ewels/MultiQC_TestData/pull/254) repository or attached to this PR: 
- [x] Code is tested and works locally (including with `--lint` flag)
- [x] `docs/README.md` is updated with link to below
- [x] `docs/modulename.md` is created
- [x] Everything that can be represented with a plot instead of a table is a plot
- [x] Report sections have a description and help text (with `self.add_section`)
- [x] There aren't any huge tables with > 6 columns (explain reasoning if so)
- [x] Each table column has a different colour scale to its neighbour, which relates to the data (eg. if high numbers are bad, they're red)
- [x] Module does not do any significant computational work

### CODE REVIEW

**Comment by jfy133** [comment]
Example report:

[multiqc_report.zip](https://github.com/ewels/MultiQC/files/11732503/multiqc_report.zip)

Also would be cool if @pirovc (author of ganon), would have a chance to have a look/leave comments if they have time/interest!

**Comment by pirovc** [comment]
Look good to me! Thanks. Later on statistics by taxonomic level can be included, but first I'd have to output them on the classify or report logs.

**Comment by jfy133** [comment]
@ewels @vladsavelyev anything else I need to do here for a review?

**Comment by ewels** [comment]
We accept bribery in the form of biscuits 🍪 😋

We'll get to it soon, just working through the backlog 😊

**Comment by jfy133** [comment]
@pirovc I realise this _may_ be out of date now since ganon 2.0 came out - to your knowledge would any changes be required (or any additional info?)

**Comment by jfy133** [comment]
@vladsavelyev @ewels I'm sure you would check thoroughly anyway, but I've attempted to add the software version and satisfy the new linting rules, but this is my first module with those, so it would be good to check those a bit more closely when you get to this.

**Comment by pirovc** [comment]
> @pirovc I realise this _may_ be out of date now since ganon 2.0 came out - to your knowledge would any changes be required (or any additional info?)

The STDOUT output in the v2 did not change from the late 1.X versions, so it should be alright.

**Comment by jfy133** [comment]
@vladsavelyev would you prefer merge in upstream into this PR (as it's getting quite old now), or better that you do it when you get to reviewing it? Just to know whether I should keep monitoring it or not

**Comment by vladsavelyev** [comment]
Ok, finally got around merging and reviewing it.

Generally looks good! But one thing is that it adds way too many columns into the general stats table, making it too wide. So I added a separate table, and made a bunch of metrics hidden by default. I kept only two metrics visible in gen stats. I did it quite arbitrary, so feel free to suggest other metrics to show in gen stats / by default.

**Review by vladsavelyev — APPROVED** [review]

**Comment by jfy133** [comment]
I'll look tomorrow and let you know, thank you very much!!

File: multiqc/modules/ganon/ganon.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+            "reads_processed",
+            "reads_classified_pc",
+            "unique_matches_pc",
+            "multiple_matches_pc",
+            "match_to_read",
+            "reads_unclassified_pc",
+            "taxonomic_entries_reported",
+        ]:
+            general_stats_headers[k] = headers[k]
+            general_stats_headers[k]["hidden"] = k not in ["reads_classified_pc", "unique_matches_pc"]
```
Comment (jfy133) [inline multiqc/modules/ganon/ganon.py:240]:
```suggestion
            general_stats_headers[k]["hidden"] = k not in ["reads_classified_pc", "unique_matches_pc", "taxonomic_entries_reported"]
```

**Comment by jfy133** [comment]
@vladsavelyev I added one more useful column to general stats but otherwise looks great, thank you!

File: multiqc/modules/ganon/ganon.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            "multiple_matches_pc",
+            "match_to_read",
+            "reads_unclassified_pc",
+            "taxonomic_entries_reported",
+        ]:
+            general_stats_headers[k] = headers[k]
+            general_stats_headers[k]["hidden"] = k not in [
+                "reads_classified_pc", 
+                "unique_matches_pc", 
+                "taxonomic_entries_reported"
```
Comment (jfy133) [inline multiqc/modules/ganon/ganon.py:243]:
```suggestion
                "taxonomic_entries_reported",
```

File: multiqc/modules/ganon/ganon.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            "multiple_matches_pc",
+            "match_to_read",
+            "reads_unclassified_pc",
+            "taxonomic_entries_reported",
+        ]:
+            general_stats_headers[k] = headers[k]
+            general_stats_headers[k]["hidden"] = k not in [
+                "reads_classified_pc", 
+                "unique_matches_pc", 
+                "taxonomic_entries_reported",
```
Comment (jfy133) [inline multiqc/modules/ganon/ganon.py:243]:
```suggestion
                "reads_classified_pc",
                "unique_matches_pc",
                "taxonomic_entries_reported",
```

**Pull request merged**